apps/els_lsp/src/els_dt_document.erl

```diff
@@ -201,3 +214,24 @@ wrapping_functions(Document, Line, Column) ->
 wrapping_functions(Document, Range) ->
   #{start := #{character := Character, line := Line}} = Range,
   wrapping_functions(Document, Line, Character).
+
+-spec find_candidates(binary()) -> [uri()].
+find_candidates(Pattern) ->
+  %% ets:fun2ms(fun(#els_dt_document{source = Source, uri = Uri, text = Text})
+  %% when Source =/= otp -> {Uri, Text} end).
+  MS = [{#els_dt_document{ uri = '$1'
+                         , source = '$2'
+                         , kind = '_'
+                         , text = '$3'
+                         , md5 = '_'
+                         ,pois = '_'}
+        , [{'=/=', '$2', otp}]
+        , [{{'$1', '$3'}}]}],
+  All = ets:select(name(), MS),
+  Fun = fun({Uri, Text}) ->
+            case binary:matches(Text, Pattern) of
```

Comment:
The main issue I see is that if we are looking for references to something like
```
to() -> paris.
```
this matches to all kind of text and atoms since the there is no tokenisation going on
```
"string containing to"
atom_containing_to
```

I think doing some cheap tokenisation would cut down false positives significantly for these cases.